app/src/main/java/com/sopt/now/LoginActivity.kt

```diff
@@ -0,0 +1,71 @@
+package com.sopt.now
+
+import UserData
+import android.content.Intent
+import android.os.Bundle
+import android.widget.Toast
+import androidx.activity.enableEdgeToEdge
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.view.ViewCompat
+import androidx.core.view.WindowInsetsCompat
+import com.sopt.now.databinding.ActivityLoginBinding
+
+class LoginActivity : AppCompatActivity() {
+    private lateinit var binding: ActivityLoginBinding
+    private val users : MutableList<UserData> = mutableListOf()
+
+    private val resultLauncher = registerForActivityResult(
+        ActivityResultContracts.StartActivityForResult()
+    ){
+        if(it.resultCode == RESULT_OK){
+            val userInfo = it.data?.getSerializableExtra("userInfo") as? UserData
+            userInfo?.let{users.add(it)}
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityLoginBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        binding.loginBtn.setOnClickListener {
+            val id = binding.idEditText.text.toString()
+            val pwd = binding.passwordEditText.text.toString()
+            val result = isLoginPossible(id, pwd)
+            if(result != null){
+                val intent = Intent(this, MainActivity::class.java)
+                intent.putExtra("login", result)
+                startActivity(intent)
+            }
+
+        }
+        binding.signUpBtn.setOnClickListener {
+            val intent = Intent(this, SignUpActivity::class.java)
+            resultLauncher.launch(intent)
+        }
+    }
+
+    private fun isLoginPossible(id: String, pwd: String): UserData? {
+        var result : UserData? = null
+        var message = ""
+        users.forEach { user ->
+            when {
+                user.id == id && user.pwd == pwd -> {
+                    result = user
+                    message = "로그인에 성공했습니다."
+                }
+                user.id == id && user.pwd != pwd -> {
+                    message = "비밀번호가 틀렸습니다."
+                }
+                user.id != id -> {
+                    message = "존재하지 않는 아이디입니다."
+                }
```

Comment:
```suggestion
                user.id != id -> {
                    message = "존재하지 않는 아이디입니다."
                }
               user.pwd != pwd -> {
                    message = "비밀번호가 틀렸습니다."
                }
                else -> {
                    result = user
                    message = "로그인에 성공했습니다."
                }
```
when 문은 최상단 조건부터 최하단 조건까지 순차적으로 검사하기 때문에 이런 식으로 변경해주는 게 좋을 것 같아요! user.id == id인 값들만 user.pwd != pwd  <- 이 조건을 검사하게 되는 걸 이용해서 중복되는 건 검사 안 할 수 있게요!